debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -112,14 +113,24 @@ public BinlogReader(String name, MySqlTaskContext context) {
         EventDeserializer eventDeserializer = new EventDeserializer() {
             @Override
             public Event nextEvent(ByteArrayInputStream inputStream) throws IOException {
-                // Delegate to the superclass ...
-                Event event = super.nextEvent(inputStream);
-                // We have to record the most recent TableMapEventData for each table number for our custom deserializers ...
-                if (event.getHeader().getEventType() == EventType.TABLE_MAP) {
-                    TableMapEventData tableMapEvent = event.getData();
-                    tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
+                try {
+                    // Delegate to the superclass ...
+                    Event event = super.nextEvent(inputStream);
+
+                    // We have to record the most recent TableMapEventData for each table number for our custom deserializers ...
+                    if (event.getHeader().getEventType() == EventType.TABLE_MAP) {
+                        TableMapEventData tableMapEvent = event.getData();
+                        tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
+                    }
+                    return event;
+                }
+                // DBZ-217 In case an event couldn't be read we create a pseudo-event for the sake of logging
+                catch(EventDataDeserializationException edde) {
+                    EventHeaderV4 header = new EventHeaderV4();
+                    header.setEventType(EventType.INCIDENT);
+                    EventData data = new EventDataDeserializationExceptionData(edde);
+                    return new Event(header, data);
```

Comment:
I agree it is a potentially risky behaviour so I think we should do two things to guard against problem

- Make the behaviour disabled by default and provide configuration option to enable it
- I've already proposed it in chat. Instead of just informing the user and throwing the event we should forward it into a special topic (like DLQ in messaging). This could be monitored by an operator and appropriate actipons can be executed.